lib/OpenQA/Utils.pm

```diff
@@ -642,6 +663,15 @@ sub render_escaped_refs {
     return bugref_to_href(url_to_href(xml_escape($text)));
 }
 
+sub find_bug_number {
+    my ($text) = @_;
+    my $marker = join('|', keys %bugrefs);
```

Comment:
This is very inefficient, you're rebuilding and compiling the regex for every function call.

Reply:
I'm aware of this - the existing functions are also inefficient and should be improved as well. In C++ one could use a `static const` variable here. I suppose the Perl equivalent would be a `local` variable?

Reply:
No, that would be a `state` variable. But in this case i would just generate the regex in the outer scope, near the definition of `%bugrefs` (which should all be uppercase as package scope variables).

Reply:
Ok, then I think you should do it in the way @kraih proposes and also (maybe in a separate commit) fix the existing functions in the same way.

Reply:
[Here's](https://github.com/mojolicious/mojo/blob/master/lib/Mojo/DOM/HTML.pm#L13) an example for how Mojolicious handles situations like this.